src/metal/scheduler.ts

```diff
@@ -25,18 +25,38 @@ let tokenCounter = 0;
 export class Frame implements FrameInterface {
   constructor(
     public timestamp: number,
-    public scrollTop: number,
     public scrollLeft: number,
+    public scrollTop: number,
+    public x: number,
+    public y: number,
     public width: number,
     public height: number
   ) {}
-  static generate(): Frame {
+  static generate(root?: Element): Frame {
```

Comment:
We probably want to have a different function for generating the Frame with a root. Having a significant branch like this on a hot code path is probably not idea. Thoughts @krisselden @chadhietala?